src/types.ts

```diff
@@ -1,4 +1,4 @@
-export type fetcherFn<Data> = ((...args: any) => Data | Promise<Data>) | null
+export type fetcherFn<Data> = (...args: any) => Data | Promise<Data>
```

Comment:
Why was this removed?

Reply:
Just saw #738, we can remove it when we implement the `passive` option (or a better name).

Reply:
because the fetcherFn type is also used in useSWRInfinite and this kind of usage is not documented as i mentioned in #738 
```js
function useSWR<Data = any, Error = any>(
  ...args:
    | readonly [keyInterface]
      // allow pass null as fetcher here
    | readonly [keyInterface, fetcherFn<Data> | null]
    | readonly [keyInterface, ConfigInterface<Data, Error>]
    | readonly [
        keyInterface,
        // allow pass null as fetcher here
        fetcherFn<Data> | null,
        ConfigInterface<Data, Error>
      ]
): responseInterface<Data, Error>
```
This still allow we pass null as fetcher and wont affect the useSWRInfinite

Reply:
I see :+1:

Then let's re-open this PR and set the merge target to the `next` branch, given this is a huge refactoring it's great to have it on v1.